src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -9231,6 +10103,62 @@ public static Vector512<float> Invoke(Vector512<float> x)
 #endif
         }
 
+        private static Vector128<float> ElementWiseSelect(Vector128<float> mask, Vector128<float> left, Vector128<float> right)
+        {
+            if (Sse41.IsSupported)
+            {
+                return Sse41.BlendVariable(left, right, ~mask);
+            }
+            else return Vector128.ConditionalSelect(mask, left, right);
+        }
+
+        private static Vector128<int> ElementWiseSelect(Vector128<int> mask, Vector128<int> left, Vector128<int> right)
+        {
+            if (Sse41.IsSupported)
+            {
+                return Sse41.BlendVariable(left, right, ~mask);
+            }
+            else return Vector128.ConditionalSelect(mask, left, right);
```

Comment:
And if BlendVariable is better here, are there other places in this file we're using ConditionalSelect we should instead be using BlendVariable?

Reply:
`ConditionalSelect` does a `bitwise select`, and thus it is logically `(mask & left) | (~mask & right)`. This is the only way to support the operation downlevel (SSE-SSSE3 capable hardware) and it is possible as a single instruction on Arm64 or on x64 hardware with AVX-512 (using `vpternlog`).

`BlendVariable` on the other hand selects `left` or `right` based solely on the most significant bit of each element. So for a 32-bit `T` it is more akin to `ConditionalSelect((mask >> 32), left, right)` (assuming signed right-shift).

`ConditionalSelect` already emits `BlendVariable` when the JIT can determine that `mask` is a per-element mask (each element is known to be `AllBitsSet` or `Zero`). But there are cases, like here, where the mask is produced earlier or modified in a way that the JIT can no longer track, but where the end user knows that it's still a per-element mask. -- We have similar `*Unsafe` APIs in a couple other places, like `ShuffleUnsafe`, where we do the generally same operation but allow for nuance in edge case handling so that it can remain efficient.

Reply:
Most if not all of the other places we're using ConditionalSelect in this file are operating on AllBitsSet/Zero.  Do we think in most/all of those cases the JIT can already see that and do the "right thing"? Or should we be switching some of those over as well?

Reply:
At a glance they were doing the right thing. I'll do another pass to see if I spot any outliers.

Long term I think we'd look at making `ConditionalSelectUnsafe` exposed publicly and integrate a suggestion to use it (if the input isn't obviously a mask) alongside https://github.com/dotnet/runtime/issues/82488.